### ISSUE TITLE
přesměrování uživatele po DNNT přihlášení

### ISSUE THREAD

**Issue by luckajirku** [ceskaexpedice/kramerius-web-client#466]
Když si uživatel otevře dokument, který je DNNT, dostane info, že se musí přihlásit, to udělá - ale po přihlášení ho to hodí do hledání v DK a zas si musí ten dokument najít. Je to takhle schválně? Přišlo by nám lepší, kdyby ho to vrátilo zpátky do toho dokumentu, který chtěl prohlížet a kvůli kterému se do DNNT přihlásil.

**Comment by honza-rychtar** [comment]
Určitě by mělo vést zpět odkud uživatel na přihlášení klikl.

**Comment by zabak** [comment]
@honza-rychtar tohle se týká přihlašování z hlášky o dostupnosti dokumentu. V podstatě jde o to, jak do linku "musíte se přihlásit" dostat URL aktuální stránky, aby se dosadilo za konstantu https://dnnt.mzk.cz/podminky-zpristupneni?target= a mohlo vzniknout https://ndk.cz/podminky-zpristupneni?target=https://ndk.cz/uuid/uuid:de7a7380-9970-45b2-afc1-bb2edd9bf0e9

V minulosti jsi udělal úpravu, aby ta hláška o právech mohla obsahovat tento link na přihlášení: https://dnnt.mzk.cz/podminky-zpristupneni?target=https://dnnt.mzk.cz/uuid/${UUID} kterou jsme zapomněli vložit do hlášky o právech při aktualizaci klienta.

**Comment by honza-rychtar** [comment]
@zabak Ano, ověřeno, že to pořád funguje. Stačí když přidáte `/uuid/${UUID}` za link jak píšeš.
Při přihlášení rovnou z navbaru se taky uživatel vrátí odkud přišel.
Takže tohle můžeme zavřít.

**Comment by zabak** [comment]
Zde zavírám, v konfiguraci si to opravíme.

**Comment by mduda100871** [comment]
Zdravím,

mysllím, že issue je pořád otevřená, protože výše uvedené řešení to ošetří jen částečně.

Pokud uživatel není přihlášen, vejde do digitální knihovny na úroveň, kde už vidí seznam všech titulů (bez aplikace jakéhokoliv filtru) a v levém panelu klikne na informační symbol (malé `i` v kroužku) u příslušné licence, tak se mu zobrazí modální okno s bližším upřesněním dané licence a je tam zároveň odkaz na přihlášení:

![dnnt-mzk-00](https://user-images.githubusercontent.com/8750622/156734597-b124ba9b-bfed-44a5-be5a-7c2cae440bbc.png)


Ten odkaz je ale bohužel nevalidní, protože v této chvíli ještě uživatel není na úrovni konkrétního dokumentu, tudíž výše uvedený placeholder `${UUID}` není aplikován a vznikne následující odkaz:

`https://dnnt.mzk.cz/uuid/`

na který když uživatel klikne, tak končí chybou (po přihlášení ke svému domovskému IdP):

![dnnt-mzk-01](https://user-images.githubusercontent.com/8750622/156734965-b11ce264-e4c8-40e4-9fcb-228d9d1cb037.png)

MD

**Comment by mduda100871** [comment]
Dočasně/trvale jsem chybu odstranil přepisem přímo ve webserveru (Apache):

`RewriteRule ^/uuid(\/){0,1}$ "https://%{HTTP_HOST}/search" [R=301,L]`

MD

**Comment by zabak** [comment]
@honza-rychtar mohl bys upravit to expandování tak, že by to ${UUID} přidávalo i to /uuid/ a ne jen samotné uuid?

**Comment by honza-rychtar** [comment]
@zabak @mduda100871 ok, upravím to. Ale přidám k tomu jiný placeholder, aby fungoval i ten s náhradou jedna ku jedné, protože se to může použít i pro jiné účely než pro přihlášení.

**Comment by honza-rychtar** [comment]
Vyřešeno přes více placeholderů https://github.com/ceskaexpedice/kramerius-web-client/issues/436#issuecomment-1198240254

**Comment by zabak** [comment]
@mduda100871 můžu to uzavřít?
